From b098f33692d75d184a3ab62095c376fd0e52d880 Mon Sep 17 00:00:00 2001
Message-Id: <b098f33692d75d184a3ab62095c376fd0e52d880.1652415137.git.geliang.tang@suse.com>
From: Kui-Feng Lee <kuifeng@fb.com>
Date: Mon, 13 Dec 2021 19:59:30 -0800
Subject: [PATCH] tools/perf: Stop using bpf_object__find_program_by_title API.
Git-commit: b098f33692d75d184a3ab62095c376fd0e52d880
Patch-mainline: v5.17-rc1
References: bsn#231
Modified-by-SEL: No

bpf_obj__find_program_by_title() in libbpf is going to be deprecated.
Call bpf_object_for_each_program to find a program in the section with
a given name instead.

Signed-off-by: Kui-Feng Lee <kuifeng@fb.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20211214035931.1148209-4-kuifeng@fb.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/perf/builtin-trace.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 0b52e08e558e..97121fb45842 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -3257,10 +3257,21 @@ static void trace__set_bpf_map_syscalls(struct trace *trace)
 
 static struct bpf_program *trace__find_bpf_program_by_title(struct trace *trace, const char *name)
 {
+	struct bpf_program *pos, *prog = NULL;
+	const char *sec_name;
+
 	if (trace->bpf_obj == NULL)
 		return NULL;
 
-	return bpf_object__find_program_by_title(trace->bpf_obj, name);
+	bpf_object__for_each_program(pos, trace->bpf_obj) {
+		sec_name = bpf_program__section_name(pos);
+		if (sec_name && !strcmp(sec_name, name)) {
+			prog = pos;
+			break;
+		}
+	}
+
+	return prog;
 }
 
 static struct bpf_program *trace__find_syscall_bpf_prog(struct trace *trace, struct syscall *sc,
-- 
2.34.1

